core/editor/src/document/document_message_handler.rs

```diff
@@ -638,6 +641,47 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					);
 				}
 			}
+			AlignSelectedLayers(dimension, aggregate) => {
+				let selected_paths = self.selected_layers_sorted();
+				if selected_paths.len() == 0 {
+					return;
+				}
+
+				let selected_layers = selected_paths.iter().map(|path| {
+					let layer = self.active_document().document.layer(path).unwrap();
+					let point = {
+						let bounding_box = layer.bounding_box(layer.transform, layer.style).unwrap();
+						match aggregate {
+							AlignAggregate::Min => bounding_box[0],
+							AlignAggregate::Max => bounding_box[1],
+							AlignAggregate::Average => bounding_box[0].lerp(bounding_box[1], 0.5),
+						}
+					};
+					let (bounding_box_coord, translation_coord) = match dimension {
+						AlignDimension::X => (point.x, layer.transform.translation.x),
+						AlignDimension::Y => (point.y, layer.transform.translation.y),
+					};
+					(path.clone(), bounding_box_coord, translation_coord)
+				});
+
+				let bounding_box_coords = selected_layers.clone().map(|(_, bounding_box_coord, _)| bounding_box_coord);
+				let aggregated_coord = match aggregate {
+					AlignAggregate::Min => bounding_box_coords.reduce(|a, b| a.min(b)).unwrap(),
+					AlignAggregate::Max => bounding_box_coords.reduce(|a, b| a.max(b)).unwrap(),
+					AlignAggregate::Average => bounding_box_coords.sum::<f64>() / selected_paths.len() as f64,
+				};
+				for (path, bounding_box_coord, translation_coord) in selected_layers {
+					let new_coord = match aggregate {
+						AlignAggregate::Min => aggregated_coord - (bounding_box_coord - translation_coord),
```

Comment:
Maybe break the triple-repeated `bounding_box_coord - translation_coord)` out into a variable?

Reply:
Ah, you made me realize that all this math is redundant haha.